src/tracer.js

```diff
@@ -267,7 +272,14 @@ export default class Tracer {
             throw new Error(`Unsupported format: ${format}`);
         }
 
-        injector.inject(spanContext, carrier);
+        if (spanContext.context) {
+            spanContext = spanContext.context();
+        }
+
+        if (spanContext) {
```

Comment:
That might be true, but opentracing compatibility passes null into this function, so we would have to change that first.